contracts/src/token/erc721/extensions/metadata.rs

```diff
@@ -53,8 +49,57 @@ impl IErc721Metadata for Erc721Metadata {
     fn symbol(&self) -> String {
         self._metadata.symbol()
     }
+}
 
-    fn base_uri(&self) -> String {
```

Comment:
In this case `#[interface_id]` won't be that helpful. Because methods inside `IErc721Metadata` trait are not same as in solidity (`token_uri` is missing). So computed interface id will be different..

Reply:
If this is the only thing preventing this PR I would say we don't expose `#[interface_id]` for the metadata extension(since I don't think it is part of the EIP itself).
One thing we should do is add in the Anotra docs or at the top of the contract that people should expose `token_uri` manually(I know it is on the internal function description, but still)